### PR TITLE
day1

### PR DESCRIPTION
**Pull request by nerisa**
@leapfrog-dipendra 
@adhpawal 
@bhuwang @Satish-Gupta @Msanjib @utsabneupane @anishmanandhar

### CODE REVIEW

File: java_training/src/day1/SearchCity.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @author nerisa
+ *
+ */
+public class SearchCity {
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		String[] cities;
```
Comment (adhpawal) [inline java_training/src/day1/SearchCity.java]: `String[] cities=new String[10];` use instead

File: java_training/src/day1/SearchCity.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		String searchString=new String();
+		Scanner userInput=new Scanner(System.in);
+		System.out.println("Enter the name of 10 cities");
+		for(int i=0;i<10;i++){
+			cities[i]=userInput.nextLine().trim();
+		}
+		System.out.println("Enter the search string");
+		searchString=userInput.nextLine();
+		for(int i=0;i<10;i++){
+			if(cities[i].toLowerCase().contains(searchString.toLowerCase())){
```
Comment (adhpawal) [inline java_training/src/day1/SearchCity.java]: use _equalsIgnoreCase_ instead

File: java_training/src/day1/SearchCity.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+				System.out.println(cities[i]);
```
Comment (adhpawal) [inline java_training/src/day1/SearchCity.java]: Add a proper message for searhc result. And , for the worst case scenario that is no Search result add a different message.

File: java_training/src/day1/SearchCity.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+public class SearchCity {
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		String[] cities;
+		cities=new String[10];
+		String searchString=new String();
```
Comment (adhpawal) [inline java_training/src/day1/SearchCity.java]: Check this out : [Avoid using new String() if possible](http://stackoverflow.com/questions/8680344/why-new-keyword-not-needed-for-string)

File: java_training/src/day1/IntegerDemo.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	public static void main(String[] args) {
+		Scanner userInput = new Scanner(System.in);
+		int valueOutRange;
+		System.out.println("Enter an integer out of its range");
+
+		try {
+			valueOutRange = userInput.nextInt();
+			System.out.println(valueOutRange);
+		} catch (InputMismatchException e) {
+			e.printStackTrace();
```
Comment (adhpawal) [inline java_training/src/day1/IntegerDemo.java]: [Why is exception.printStackTrace() considered bad practice?](http://stackoverflow.com/questions/7469316/why-is-exception-printstacktrace-considered-bad-practice)

You decide.

File: java_training/src/day1/ComputeInitial.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class ComputeInitial {
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		Scanner userInput=new Scanner(System.in);
+		String name=new String();
+		//String initial=new String();
```
Comment (adhpawal) [inline java_training/src/day1/ComputeInitial.java]: Remove the commented lines of code.

File: java_training/src/day1/FullNameConcat.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		String salutation=new String();
+		String firstName=new String();
+		String middleName=new String();
+		String lastName=new String();
+		StringBuilder fullName=new StringBuilder();
+		Scanner inputValue=new Scanner(System.in);
+		do{
+			System.out.println("Enter your salutation");
+			salutation=inputValue.nextLine().trim();
+			fullName.append(salutation);
```
Comment (adhpawal) [inline java_training/src/day1/FullNameConcat.java]: So, User just press *Enter Enter Enter .... \* a zillion time and `fullName.append(salutation);` will be executed the same zillion time.

**Comment by adhpawal** [comment]
Please Use proper Reasonable Commit message. You have following Commit Message
    - first push
    - changed class name
    - initialized
    - stringSeperator variable added

It is a learning curve and i hope you will do better in it . But you need to learn the importance of Commit Message on long run.

And , You should make a habit of commiting frequently.

File: java_training/src/day1/SearchCity.java (hunk 2 above)
Comment (nerisa) [inline java_training/src/day1/SearchCity.java]: @adhpawal The program searches the city name that contains a certain substring. For e.g. if there's Kathmandu the typing Ka only returns the Kathmandu

File: java_training/src/day1/SearchCity.java (hunk 1 above)
Comment (amannepid) [inline java_training/src/day1/SearchCity.java]: @nerisa Please remove unused and irrelevant method description.

File: java_training/src/day1/SearchCity.java (hunk 2 above)
Comment (amannepid) [inline java_training/src/day1/SearchCity.java]: @nerisa  why not String[] cities = new String[10]; ?

File: java_training/src/day1/SearchCity.java (hunk 1 above)
Comment (nerisa) [inline java_training/src/day1/SearchCity.java]: @leapfrog-dipendra dai i didnt understand your comment. Which method are you referring to?

**Pull request merged**